src/Relation/Binary/Lattice.agda

```diff
@@ -243,3 +298,108 @@ record BoundedLattice c ℓ₁ ℓ₂ : Set (suc (c ⊔ ℓ₁ ⊔ ℓ₂)) wher
   boundedMeetSemilattice : BoundedMeetSemilattice c ℓ₁ ℓ₂
   boundedMeetSemilattice = record
     { isBoundedMeetSemilattice = isBoundedMeetSemilattice }
+
+  lattice : Lattice c ℓ₁ ℓ₂
+  lattice = record { isLattice = isLattice }
+
+  open Lattice lattice
+    using (joinSemilattice; meetSemilattice; poset; preorder)
+    public
+
+------------------------------------------------------------------------
+-- exponential
+--
+-- Heyting algebra is bounded lattice with exponential.
+
+Exponential : ∀ {a ℓ} {A : Set a} → Rel A ℓ → Op₂ A → Op₂ A → Set _
```

Comment:
I don't have a strong opinion on the name, but the definition should go at the top of the file with `Infinimum` and `Supremum`